### PR TITLE
chore(root): Release version `2.1.0` for all applications

### PR DESCRIPTION
**Pull request by rifont**
### What changed? Why was the change needed?
* Release version `2.1.0` for all applications
* Add `type:app` tags to application package nx configuration to simplify release flow
  * make tag configuration consistent following NX recommended naming (apps: `tag: "type:app"`, packages: `tag: "package:public"` -> `tag: "type:package"` - all packages are public and non-public packages are libraries, so there's no need to be explicit about a package being public)
* Organize releasable artifacts into NX groups in nx.json

### Screenshots

<details>
<summary><strong>Expand for optional sections</strong></summary>

### Special notes for your reviewer

</details>

### CODE REVIEW

File: scripts/release.mjs [hunk 1]
```diff
@@ -29,13 +29,15 @@ import { execa } from 'execa';
   }
 
   const { workspaceVersion, projectsVersionData } = await releaseVersion({
+    projects: ['tag:type:package'],
```
Comment (rifont) [inline scripts/release.mjs:41]: @SokratisVidros is this script intended to release packages only? If not, we need to remove this filter

File: nx.json [hunk 2]
```diff
@@ -21,24 +21,46 @@
...
+          "@novu/react-native",
+          "@novu/nextjs",
+          "@novu/node",
+          "@novu/providers",
+          "@novu/shared",
+          "@novu/stateless"
+        ],
+        "projectsRelationship": "independent",
+        "version": {
+          "generatorOptions": {
```
Comment (SokratisVidros) [inline nx.json:62]: Let's add `"conventionalCommits": true,` here.

File: scripts/release.mjs (hunk 1 above)
Comment (SokratisVidros) [reply scripts/release.mjs:41]: It releases only packages.

File: scripts/release.mjs [hunk 3]
```diff
@@ -29,13 +29,15 @@ import { execa } from 'execa';
...
     specifier,
     dryRun,
     verbose,
     firstRelease: true,
```
Comment (SokratisVidros) [inline scripts/release.mjs]: This was committed accidentally. Please remove.

**Review by SokratisVidros — APPROVED** [review]

**Pull request merged**